_aicspylibczi/Reader.cpp

```diff
@@ -235,37 +254,33 @@ namespace pylibczi {
       }
       SubblockSortable subblocksToFind(&plane_coord_, index_m_, isMosaic());
       SubblockIndexVec matches = getMatches(subblocksToFind);
-      ImageVector images;
-      images.reserve(matches.size()); // this will under-reserve in the case of BGR images
-      bool bgrFlag=false;
+      m_pixelType = matches.front().first.pixelType();
+      size_t bgrScaling = ImageFactory::numberOfChannels(m_pixelType);
 
+      libCZI::IntRect w_by_h = getSceneYXSize();
+      size_t n_of_pixels = matches.size()*w_by_h.w*w_by_h.h*bgrScaling;
+      ImageFactory imageFactory(m_pixelType, n_of_pixels);
+
+      imageFactory.setMosaic(isMosaic());
+      size_t memOffset = 0;
       for_each(matches.begin(), matches.end(), [&](const SubblockIndexVec::value_type& match_) {
           auto subblock = m_czireader->ReadSubBlock(match_.second);
           const libCZI::SubBlockInfo& info = subblock->GetSubBlockInfo();
-          auto image = ImageFactory::constructImage(subblock->CreateBitmap(),
-              &info.coordinate, info.logicalRect, info.mIndex);
-          // This was conditional on split_bgr_ but that's a bad idea so I'm removing it.
-          // bgr images will always be split into their base single channel types brg24 => uint8_t
-          if (ImageFactory::numberOfChannels(image->pixelType())>1) {
-              if ( bgrFlag ){
-                  throw ImageAccessUnderspecifiedException(0, 1,
-                      "In a multi-channel BGR image C must be explicitly specified. This is to avoid confusion between BGR expanded channels.");
-              }
-              auto splitImages = ImageFactory::splitToChannels(image);
-              for_each(splitImages.begin(), splitImages.end(), [&images](Image::ImVec::value_type& image_) { images.push_back(image_); });
-              bgrFlag = true;
-          }
-          else
-              images.push_back(image);
+          if (m_pixelType!=info.pixelType)
+              throw PixelTypeException(info.pixelType, "Selected subblocks have inconsistent PixelTypes."
+                                                       " You must select subblocks with consistent PixelTypes.");
+
+          auto image = imageFactory.constructImage(subblock->CreateBitmap(),
```

Comment:
there is something strange about this calling constructImage and returning an image and then never using it.  It has to be a naming problem, I suppose.  What is constructImage really doing?

Reply:
It's actually internally adding it to a vector but returns it to the user in case they want to do something with it. I don't have to assign the return value? What's your preferred behavior here?

Reply:
I guess the preferred thing is to drop the `auto image = ` part of the line.